packages/core/src/amazonq/util/telemetryHelper.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MetricName } from '../../shared/telemetry/clienttelemetry'
+
+// TODO: technically this can also be any metric name that occurs in the flow now
+type Event = 'initialRequestTime' | 'webviewToDispatcher' | 'dispatcherToFeature' | 'finalRequestTime' | MetricName
+
+export class AmazonQUITelemetryHelper {
```

Comment:
This class feels like it can be generalized with the use of some generics

```
export class BaseTelemetryHelper<Data = { event: Event, time: number}, tabIDOrConversationId: string>
```
where the names and types in the generics wouldn't actually be those, but just to map your current implementation for this example.

And then also `UIMetrics` would be something else like `Datas` (the name sucks but something like this).

Then the whole thing could live in spans.ts and then if any other feature wants to be able to use the `listen` functionality they can make a quick subclass of the implementation for their own feature.